### ISSUE TITLE
Wallet Cards wont show up on new phone

### ISSUE THREAD

**Issue by rom4ster** [KieronQuinn/ClassicPowerMenu#171]
So on my phone it keeps saying that there was an error getting your cards and the culprit seems to be this specific code in CPMQuickAccessWalletServiceInfo
```

   resolveInfos = context.getPackageManager().queryIntentServices(intent,
                    PackageManager.MATCH_DEFAULT_ONLY);
```

On my old phone where it works, this returns something but on my new phone no return. I made some more code to just search through the services based on the keyword "tapandpay"

![image](https://github.com/user-attachments/assets/8396bfc4-60b2-4177-bea9-00137bd8b297)

On the left you can see the result on my old phone and on the right my new phone. Notice the WalletQuickAccess is gone? 

I went and checked GMS version and its not only the same size but the same version so I am confused as to what is happening. 

I do know that on japan roms (im on ANDROID 14 btw) Osaifu Keitai is present and the nfc tech is not exactly the same (not the phone is a japanese unlocked model with jp firmware). 

So perhaps this is an issue with it being a Japan rom? The one dude on discord did not want to test so I dont know anyone else who has a Xperia 1 VI so idk if anyone has other jp phones or can otherwise reproduce this or have a solution for this?

**Comment by KieronQuinn** [comment]
If the QAW service is gone, there is unfortunately nothing that can be done. Google don't have any obligation to keep it, since they're moving away from using it. The wallet shortcut on the lock screen has launched the main wallet app for a while now, and the QAW implementing SystemUI activity has never added support for edge to edge - strong indication it's abandoned.

In short, it's over.

**Comment by rom4ster** [comment]
> If the QAW service is gone, there is unfortunately nothing that can be done. Google don't have any obligation to keep it, since they're moving away from using it. The wallet shortcut on the lock screen has launched the main wallet app for a while now, and the QAW implementing SystemUI activity has never added support for edge to edge - strong indication it's abandoned.
> 
> In short, it's over.

This makes sense however, one of the problems is that failing on that method also fails the loyalty cards. Loyalty cards dont seem to use the same system and smartspacer loyalty integration works. I wonder if we can fail just the payment cards and not the loyalty cards. This would allow for *almost* full experience even if QAW is gone since the app did not add functionality outside switching the card. 

This issue may still be something weird on jp roms as my wallet QS tile says "not available" .

**Comment by KieronQuinn** [comment]
> > If the QAW service is gone, there is unfortunately nothing that can be done. Google don't have any obligation to keep it, since they're moving away from using it. The wallet shortcut on the lock screen has launched the main wallet app for a while now, and the QAW implementing SystemUI activity has never added support for edge to edge - strong indication it's abandoned.
> > In short, it's over.
> 
> This makes sense however, one of the problems is that failing on that method also fails the loyalty cards. Loyalty cards dont seem to use the same system and smartspacer loyalty integration works. I wonder if we can fail just the payment cards and not the loyalty cards. This would allow for _almost_ full experience even if QAW is gone since the app did not add functionality outside switching the card.
> 
> This issue may still be something weird on jp roms as my wallet QS tile says "not available" .

Loyalty cards get appended onto the end of the regular cards, so it requires the service to be there. No regular cards service, no loyalty cards.

The QS tile showing not available will be for this reason too, yes.

**Comment by rom4ster** [comment]
> > > If the QAW service is gone, there is unfortunately nothing that can be done. Google don't have any obligation to keep it, since they're moving away from using it. The wallet shortcut on the lock screen has launched the main wallet app for a while now, and the QAW implementing SystemUI activity has never added support for edge to edge - strong indication it's abandoned.
> > > In short, it's over.
> > 
> > 
> > This makes sense however, one of the problems is that failing on that method also fails the loyalty cards. Loyalty cards dont seem to use the same system and smartspacer loyalty integration works. I wonder if we can fail just the payment cards and not the loyalty cards. This would allow for _almost_ full experience even if QAW is gone since the app did not add functionality outside switching the card.
> > This issue may still be something weird on jp roms as my wallet QS tile says "not available" .
> 
> Loyalty cards get appended onto the end of the regular cards, so it requires the service to be there. No regular cards service, no loyalty cards.
> 
> The QS tile showing not available will be for this reason too, yes.

What you are saying makes sense but my point of confusion is that smartspacer loyalty cards work just fine, so is it just the way CPM is written or is it deeper than that?

**Comment by KieronQuinn** [comment]
> > > > If the QAW service is gone, there is unfortunately nothing that can be done. Google don't have any obligation to keep it, since they're moving away from using it. The wallet shortcut on the lock screen has launched the main wallet app for a while now, and the QAW implementing SystemUI activity has never added support for edge to edge - strong indication it's abandoned.
> > > > In short, it's over.
> > > 
> > > 
> > > This makes sense however, one of the problems is that failing on that method also fails the loyalty cards. Loyalty cards dont seem to use the same system and smartspacer loyalty integration works. I wonder if we can fail just the payment cards and not the loyalty cards. This would allow for _almost_ full experience even if QAW is gone since the app did not add functionality outside switching the card.
> > > This issue may still be something weird on jp roms as my wallet QS tile says "not available" .
> > 
> > 
> > Loyalty cards get appended onto the end of the regular cards, so it requires the service to be there. No regular cards service, no loyalty cards.
> > The QS tile showing not available will be for this reason too, yes.
> 
> What you are saying makes sense but my point of confusion is that smartspacer loyalty cards work just fine, so is it just the way CPM is written or is it deeper than that?

Smartspacer has zero dependencies on the service, since it doesn't show regular cards

**Comment by rom4ster** [comment]
Ok I took a look at the code and now I think I fully understand the depth of the problem. The QAW service is not just for payment cards but also loyalty cards in CPM. This makes sense since both are supported by CPM. This probably also explains why smartspacer supports popouts still. It probably is a more flexible solution to not use QAW for loyalty cards but from my investigation (essentially I tried to implement that) doing so is a massive undertaking, probably neccessitating a rewrite. 

I think that it would be cool however I am in no position to do a rewrite of this. I am not quite well versed with android views and what not so it would take me a long time (I only know compose lol). 

I am clearly not the man for this job at this point so I have just added a tasker shortcut to take me to the wallet app.

If that widgets suggestion ever becomes a thing though then I could just use smartspacer 

but yeah, I thought I would post the summary just so if anyone does want this challenge they know the full story.
